gravitee-am-gateway/gravitee-am-gateway-handler/gravitee-am-gateway-handler-core/src/main/java/io/gravitee/am/gateway/handler/root/resources/endpoint/mfa/MFAEnrollEndpoint.java

```diff
@@ -111,6 +110,10 @@ private void renderPage(RoutingContext routingContext) {
                 routingContext.fail(401);
                 return;
             }
+            var context = new MfaFilterContext(routingContext, routingContext.get(ConstantKeys.CLIENT_CONTEXT_KEY), factorManager);
+            if (context.userHasMatchingFactors()) {
+                redirectToAuthorize(routingContext);
```

Comment:
missing `return;` after the `redirectToAuthorize(routing);` 

```Exception in thread "Thread-15" java.lang.IllegalStateException: Response head already sent
	at io.vertx.core.http.impl.Http1xServerResponse.checkHeadWritten(Http1xServerResponse.java:703)
	at io.vertx.core.http.impl.Http1xServerResponse.putHeader(Http1xServerResponse.java:193)
	at io.vertx.core.http.impl.Http1xServerResponse.putHeader(Http1xServerResponse.java:68)
	at io.vertx.rxjava3.core.http.HttpServerResponse.putHeader(HttpServerResponse.java:279)
	at io.gravitee.am.gateway.handler.root.resources.endpoint.AbstractEndpoint.lambda$renderPage$1(AbstractEndpoint.java:77)

Reply:
added with an additional warn log